packages/taco/src/viem-taco.ts

```diff
@@ -0,0 +1,138 @@
+import { ThresholdMessageKit } from '@nucypher/nucypher-core';
+import { type Account, Domain, type PublicClient } from '@nucypher/shared';
+import { ethers } from 'ethers';
+
+import { Condition } from './conditions/condition';
+import { ConditionContext } from './conditions/context';
+import { decrypt as ethersDecrypt, encrypt as ethersEncrypt } from './taco';
+import {
+  createTacoCompatibleProvider,
+  createTacoCompatibleSigner,
+} from './wrappers';
+
+/**
+ * Encrypts a message under given conditions using viem clients.
+ *
+ * This is a viem-compatible version of the encrypt function that accepts viem clients
+ * instead of ethers providers and signers.
+ *
+ * @export
+ * @param {PublicClient} viemPublicClient - Viem PublicClient for network operations
+ * @param {Domain} domain - Represents the logical network for encryption (must match ritualId)
+ * @param {Uint8Array | string} message - The message to be encrypted
+ * @param {Condition} condition - Condition under which the message will be encrypted
+ * @param {number} ritualId - The ID of the DKG Ritual to be used for encryption
+ * @param {Account} viemAuthSigner - The viem account that will be used to sign the encrypter authorization
+ *
+ * @returns {Promise<ThresholdMessageKit>} Returns Promise that resolves with an instance of ThresholdMessageKit
+ *
+ * @throws {Error} If the active DKG Ritual cannot be retrieved an error is thrown
+ *
+ * @example
+ * ```typescript
+ * import { createPublicClient, createWalletClient, http } from 'viem';
+ * import { polygonAmoy } from 'viem/chains';
+ * import { privateKeyToAccount } from 'viem/accounts';
+ *
+ * // Using explicit clients
+ * const viemPublicClient = createPublicClient({
+ *   chain: polygonAmoy,
+ *   transport: http()
+ * });
+ *
+ * const viemAuthSigner = privateKeyToAccount('0x...');
+ *
+ * const encrypted = await encryptWithViem(
+ *   viemPublicClient, // Your viem public client
+ *   'lynx',           // TACo domain
+ *   'Hello, secret!', // Message to encrypt
+ *   condition,        // Access condition
+ *   27,               // Ritual ID
+ *   viemAuthSigner    // Your viem account
+ * );
+ * ```
+ */
+export const encryptWithViem = async (
```

Comment:
i'm not sure I like this api design. Is there any way add this functionality with only 1 encrypt function? maybe by injecting the viem provider?